node/src/sub_lib/neighborhood.rs

```diff
@@ -502,16 +502,53 @@ pub enum ExpectedService {
     Nothing,
 }
 
+impl ExpectedService {
+    pub fn exit_node_key_opt(&self) -> Option<PublicKey> {
+        match self {
+            ExpectedService::Exit(key, _, _) => Some(key.clone()),
+            _ => None,
+        }
+    }
+
+    pub fn public_key_opt(&self) -> Option<PublicKey> {
+        match self {
+            ExpectedService::Exit(key, _, _) | ExpectedService::Routing(key, _, _) => {
+                Some(key.clone())
+            }
+            _ => None,
+        }
+    }
+
+    pub fn wallet_opt(&self) -> Option<&Wallet> {
+        match self {
+            ExpectedService::Exit(_, wallet, _) | ExpectedService::Routing(_, wallet, _) => {
+                Some(wallet)
+            }
+            _ => None,
+        }
+    }
+
+    pub fn rate_pack_opt(&self) -> Option<&RatePack> {
+        match self {
+            ExpectedService::Exit(_, _, rate_pack) | ExpectedService::Routing(_, _, rate_pack) => {
+                Some(rate_pack)
+            }
+            _ => None,
+        }
+    }
+}
+
 #[derive(Clone, Debug, PartialEq, Eq)]
 pub enum ExpectedServices {
     OneWay(Vec<ExpectedService>),
-    RoundTrip(Vec<ExpectedService>, Vec<ExpectedService>, u32),
+    RoundTrip(Vec<ExpectedService>, Vec<ExpectedService>),
 }
 
 #[derive(Clone, Debug, PartialEq, Eq)]
 pub struct RouteQueryResponse {
     pub route: Route,
     pub expected_services: ExpectedServices,
+    pub hostname_opt: Option<String>,
```

Comment:
This hostname now comes in in the `RouteQueryRequest` and is turned around into the response. That's the easiest way to pass the hostname from the code that makes the request to the code that receives the response, since it can no longer be stored in the `ProxyServer` with the RRID.